### PR TITLE
fix: google sheet ui issues, removed redundant code

### LINKED ISSUE #18032: While creating a gsheet datasource, creation of the query should only be possible on valid datasource
design: https://www.figma.com/file/Qs7p5frNx7uEfVYpADGk0d/Query-Pane-Improvements?node-id=1132%3A45433

**Create Mode**
Current behaviour: when I click `save and authorize`, save happens first, then the user is shown the auth popup in a separate page. There are two possibilities here:
1. auth success: no problem, the datasource is saved with the successful auth, user sees the datasource overview page
2. auth failure: the datasource is saved without auth

Proposed change: 
1. auth success: no problem, the datasource is saved with the successful auth, show the datasource overview page
2. auth failure: _The datasource should be saved as it is today, but on the view datasource page, I should see an error banner as shown in the design, and the `New API` button should be disabled_

### PR DESCRIPTION
**Pull request by AmanAgarwal041**
## Description

This Pr deals with :
- Google Sheets UI fixes
- Limiting users to build new query if the Google Sheet Datasource is unauthorized
- Removal of some redundant code and moving it to one place to update the ui designs in future easily

Fixes #18032

Media
[Loom Video for Google Sheet Fixes](https://www.loom.com/share/4041022f76c341f68ffd5da7dbb0cebc)

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

- Manual
- Cypress

### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: app/client/src/pages/Editor/SaaSEditor/DatasourceForm.tsx [hunk 1]
```diff
@@ -277,24 +285,37 @@ class DatasourceSaaSEditor extends JSONtoForm<Props, State> {
...
+                          params: {
+                            viewMode: false,
+                          },
+                        }),
+                      );
+                    }}
+                    text="EDIT"
+                  />
+                  {isGoogleSheetPlugin && (
+                    <NewActionButton
```
Comment (ankitakinger) [inline app/client/src/pages/Editor/SaaSEditor/DatasourceForm.tsx:308]: @berzerkeer I think this is a new button getting added here, please check if need any permission driven views to be added for this as well.

Comment (berzerkeer) [reply app/client/src/pages/Editor/SaaSEditor/DatasourceForm.tsx:308]: @ankitakinger yes it needs create page actions on page and create datasource actions on datasource.

Comment (AmanAgarwal041) [reply app/client/src/pages/Editor/SaaSEditor/DatasourceForm.tsx:316]: @ankitakinger @berzerkeer Added the create page actions and datasource actions flag on the new action button.

**Review by berzerkeer — DISMISSED** [review]

**Review by sneha122 — DISMISSED** [review]

File: app/client/src/pages/Editor/DataSourceEditor/NewActionButton.tsx [hunk 2]
```diff
@@ -88,13 +90,18 @@ function NewActionButton(props: NewActionButtonProps) {
...
       iconPosition={IconPositions.left}
       isLoading={isSelected || props.isLoading}
-      onClick={createQueryAction}
+      onClick={disabled ? noop : createQueryAction}
+      style={style}
       tag="button"
-      text={pluginType === PluginType.DB ? "New Query" : "New API"}
+      text={
+        pluginType === PluginType.DB || pluginType === PluginType.SAAS
+          ? "New Query"
```
Comment (Irongade) [inline app/client/src/pages/Editor/DataSourceEditor/NewActionButton.tsx:102]: Can these strings be made constants and added to the `constants/messages.ts` file.

File: app/client/src/pages/Editor/IntegrationEditor/DatasourceCard.tsx [hunk 3]
```diff
@@ -319,26 +320,32 @@ function DatasourceCard(props: DatasourceCardProps) {
...
+                      ? "t--generate-template"
+                      : "t--reconnect-btn"
+                  }
+                  onClick={
+                    datasource.isConfigured
+                      ? routeToGeneratePage
+                      : editDatasource
+                  }
+                  text={
+                    datasource.isConfigured ? "GENERATE NEW PAGE" : "RECONNECT"
```
Comment (Irongade) [inline app/client/src/pages/Editor/IntegrationEditor/DatasourceCard.tsx:338]: Here as well. All exposed strings could be added to the messages file.

Comment (sneha122) [reply app/client/src/pages/Editor/IntegrationEditor/DatasourceCard.tsx]: @Irongade Resolved. Please check

File: app/client/src/pages/Editor/DataSourceEditor/NewActionButton.tsx (hunk 2 above)
Comment (sneha122) [reply app/client/src/pages/Editor/DataSourceEditor/NewActionButton.tsx]: @Irongade Resolved. Please check

**Review by Irongade — APPROVED** [review]

**Pull request merged**